tests/virt_autotest/sriov_network_card_pci_passthrough.pm

```diff
@@ -187,12 +175,24 @@ sub find_sriov_ethernet_devices {
     my @sriov_devices;
     foreach (@nic_devices) {
         if ((script_run "lspci -v -s $_ | grep -q 'SR-IOV'") == 0) {
-            #only those vfs whose pv can be brought up can be passed through to guest vms
+            # Any VFs can be passed through to guests
+            # But only those VFs whose pv has physical network connection can get an IP from DHCP server
             my $nic = script_output "ls -l /sys/class/net |grep $_ | awk '{print \$9}'";
-            script_run "echo \"BOOTPROTO='none'\" > /etc/sysconfig/network/ifcfg-$nic" unless $nic eq get_var('SUT_NETDEVICE', 'eth0');
-            push @sriov_devices, $_ if (script_run("ifup $nic") == 0);
+            if ($nic eq get_var('SUT_NETDEVICE', 'eth0')) {
+                push @sriov_devices, $_;
+                record_info("Find SR-IOV devices", "$_    $nic");
+            }
+            else {
+                assert_script_run("ip link set $nic up");
+                if (script_output("cat /sys/class/net/$nic/carrier") eq '1') {
+                    push @sriov_devices, $_;
+                    record_info("Find SR-IOV devices", "$_    $nic");
+                }
+            }
```

Comment:
I think just 
```
assert_script_run("ip link set $nic up");
if (script_output("cat /sys/class/net/$nic/carrier") eq '1') {
    push @sriov_devices, $_;
    record_info("Find SR-IOV devices", "$_    $nic");
}
```
is enough.

Reply:
The tests use the master interface(SUT_NETDEVICE) on most of the workers. The 'else' steps to check the physical connection are only required on few cases, ie. the zen3 machine. So I choosed duplicate codes instead of duplicate steps in runtime for efficiency.